STYLEGUIDE.md

```diff
@@ -0,0 +1,716 @@
+
+❌ Examples of **incorrect** code:
+```ts
+/* Don't use uppercase snake case */
+const FOO_BAR = 'foo';
+
+/* Don't use snake case */
+const foo_bar = 'foo';
+
+/* Don't use different case for static variables */
+class FooBar {
+  static FOO_BAR;
+}
+```
+
+✅ Examples of **correct** code:
+```ts
+/* Use camel case */
+const fooBar = 'foo';
+
+/* Use different case for static variables */
+class FooBar {
+  static fooBar;
+}
+```
+
+### Use pascal case for classes ✓
+Pascal case is common in most JavaScript/TypeScript projects for classes.
+
+✅ Examples of **correct** code:
+```ts
+/* Use pascal case */
+class FooBar {
+  ...
+}
+```
+
+### Use pascal case for interface names and don't prefix them
+Pascal case is common in most JavaScript/TypeScript projects for interfaces.
+Don't prefix interfaces with the hungarian notation 'I' (or other prefixes).
+
+❌ Examples of **incorrect** code:
+```ts
+/* Don't use 'I' prefix (or any other prefixes) */
+interface IFooBar {
+  ...
+}
+```
+
+✅ Examples of **correct** code:
+```ts
+/* Use pascal case without prefix */
+interface FooBar {
+  ...
+}
+```
+
+### Use pascal case for enums and enum properties
+Pascal case is common in most JavaScript/TypeScript projects for enums and enum properties.
+
+❌ Examples of **incorrect** code:
+```ts
+/* Don't use upper case snake case for enums and enum properties */
+enum SomeEnum {
+  FOO_BAR,
+  BAR_FOO
+}
+
+/* Don't use camel case for enums and enum properties */
+enum SomeEnum {
+  fooBar,
+  barFoo
+}
+```
+
+✅ Examples of **correct** code:
+```ts
+/* Use pascal case */
+enum SomeEnum {
+  FooBar,
+  BarFoo
+}
+```
+### Use verbs in function names
+Every function should do something and that action should be described with a verb.
+The verb should fit the return type and value.
+An exception to this can be methods, that operate on an instance directly, e. g. `instance.toString()`, `instance.asObject()`.
+
+❌ Examples of **incorrect** code:
+```ts
+/* Don't use functions without verb */
+function toFoo(): Foo {
+  ...
+}
+
+/* Don't use functions with misleading verbs */
+function isFoo(): void {
+  ...
+  throw Error(...);
+}
+
+/* Don't use camel case for enums and enum properties */
+enum SomeEnum {
+  fooBar,
+  barFoo
+}
+```
+
+✅ Examples of **correct** code:
+```ts
+/* Use functions with verbs */
+function transformToFoo(): Foo {
+  ...
+}
+
+/* Use functions with verbs that fit the return type */
+function isFoo(): boolean {
+  ...
+}
+
+/* Use functions with verbs that fit the return type */
+function validateFoo(): void {
+  ...
+  throw Error(...);
+}
+
+/* Exception, you can omit verbs in methods, that transform the instance that they are invoked on */
+class FooBar {
+  toString(): string {
+    ...
+  }
+
+  asObject(): Record<string, any> {
+    ...
+  }
+}
+```
+
+## Strings
+
+### Use single quotes ✓
+
+Use single quotes, unless not possible otherwise, e. g. when your string
```

Comment:
```suggestion
Use single quotes, unless not possible otherwise, e. g. when your string contains single quotes.
```

Reply:
I would change it slightly 

```suggestion
Use single quotes, unless you can avoid escaping characters otherwise, e.g. double quotes when your string contains single quotes.
```